### ISSUE TITLE
keras.ops.numpy.i0 overflows to Inf for finite float64 result near x=713

### ISSUE THREAD

**Issue by ALinrunrun** [keras-team/keras#23399]
## Description

`keras.ops.numpy.i0` returns `inf` for an input whose correct `float64` result is still finite.

For the input:

```text
713.0
```

the modified Bessel function `i0(713.0)` should be approximately:

```text
6.705128263670996e+307
```

This value is very large, but it is still within the finite `float64` range. However, `keras.ops.numpy.i0` returns `inf`.

This looks like an incorrect overflow / numerical accuracy issue in `keras.ops.numpy.i0` for large finite `float64` inputs.

## Minimal reproduction

```python
import os
os.environ["KERAS_BACKEND"] = "tensorflow"
os.environ["CUDA_VISIBLE_DEVICES"] = "-1"
os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
os.environ["TF_ENABLE_ONEDNN_OPTS"] = "0"

import warnings
warnings.filterwarnings("ignore")

import numpy as np
import keras

x = np.array([713.0], dtype=np.float64)

actual = np.asarray(
    keras.ops.numpy.i0(
        keras.ops.convert_to_tensor(x, dtype="float64"),
    ),
    dtype=np.float64,
)

expected = np.array([6.705128263670996e+307], dtype=np.float64)

print("keras:", keras.__version__)
print("backend:", keras.config.backend())
print("input:", x.tolist())
print("actual:", actual.tolist())
print("expected:", expected.tolist())
```

Output:

```text
keras: 3.14.0
backend: tensorflow
input: [713.0]
actual: [inf]
expected: [6.705128263670996e+307]
```

## Expected behavior

`keras.ops.numpy.i0` should return a finite `float64` value close to:

```text
[6.705128263670996e+307]
```

The expected value is based on a high-precision mpmath reference rounded to `float64`. Although the result is close to the upper end of the `float64` range, it is still finite.

## Actual behavior

`keras.ops.numpy.i0` currently returns:

```text
[inf]
```

The result incorrectly overflows to infinity even though the correct `float64` value is finite.

**Comment by maitry63** [comment]
@dhantule @ALinrunrun we can close this issue! As it is duplicate of https://github.com/keras-team/keras/issues/23400.
